enumerables.rb

```diff
@@ -0,0 +1,168 @@
+module Enumerable
+  # my_each
+  def my_eac
+    return enum_for(:my_each) unless block_given?
+
+    arr = self if self.class == Array
+    arr = to_a if self.class == Range || Hash
+    0.upto(arr.length - 1) do |index|
+      yield(arr[index])
+    end
+    arr
+  end
```

Comment:
#my_each when a block is given when self is a range returns the range itself after calling the given block once for each element in self

Reply:
block = proc { |num| num > 4 }
range = 5..50
range.my_each(&block) #5..50